### PR TITLE
feat: Add  a python package

### PR DESCRIPTION
**Pull request by aborgna-q**
Adds all the boilerplate for a `quantinuum-hugr` python package.

This includes conditional CI checks (dependant on the changed files), and other metadata.

What's missing is a `release-please` configuration to generate python changelogs. `release-plz` should already be configured to only include changes that modify the rust project.

This is the foundation required for #486 and https://github.com/CQCL/guppylang/issues/173, and it should make adding another package for #866 quite easy.

### CODE REVIEW

**Comment by aborgna-q** [comment]
Uh, the "approve required checks if they are skipped" doesn't quite work. If we modify files that should be checked _and also_ files to ignore, we end up running both workflows -.-

The github docs are not really helpful:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks#handling-skipped-but-required-checks

I think the best option is to just use https://github.com/dorny/paths-filter. It seems to be a widely used alternative.

File: .github/workflows/ci-rs.yml
Comment (cqc-alec) [inline .github/workflows/ci-rs.yml:81]: It may be an issue that this is a "required" check for PRs, when it is sometimes skipped. Not sure what the best fix for this is; TKET does a [hack](https://github.com/CQCL/tket/blob/5ed1c366b419e0de03c581482ef1e94d12653a12/.github/workflows/build_and_test.yml#L580) where it creates a new "required" check that is a meta-check that everything succeeded or was skipped.

Comment (aborgna-q) [reply .github/workflows/ci-rs.yml:83]: I added the meta-check. I'll need to edit the required check name in the repo config before merging this PR.

File: quantinuum-hugr-py/README.md
Comment (aborgna-q) [reply quantinuum-hugr-py/README.md:5]: I configured "codecov flags" to group the different coverages.
https://docs.codecov.com/docs/flags

I'll wait until that is active on codecov before updating this url.

File: .gitignore
Comment (aborgna-q) [reply .gitignore:1]: Done. I had merged guppy's ignore file here, now it's a bit cleaner.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**
